upstairs/src/lib.rs

```diff
@@ -2023,17 +2140,21 @@ impl BlockReq {
  * waiter that it can block on.
  */
 pub struct BlockReqWaiter {
-    recv: std_mpsc::Receiver<i32>,
+    recv: std_mpsc::Receiver<Result<(), CrucibleError>>,
 }
 
 impl BlockReqWaiter {
-    fn new(recv: std_mpsc::Receiver<i32>) -> BlockReqWaiter {
+    fn new(
+        recv: std_mpsc::Receiver<Result<(), CrucibleError>>,
+    ) -> BlockReqWaiter {
         Self { recv }
     }
 
-    pub fn block_wait(&mut self) {
-        // TODO: Instead of i32, errors should be bubbled up.
-        let _ = self.recv.recv();
+    pub fn block_wait(&mut self) -> Result<(), CrucibleError> {
+        match self.recv.recv() {
+            Ok(v) => v,
+            Err(_) => crucible_bail!(RecvDisconnected),
```

Comment:
Is the only way this recv can fail is with a disconnect?

Reply:
Yeah:

> If the corresponding Sender has disconnected, or it disconnects while this call is blocking, this call will wake up and return Err to indicate that no more messages can ever be received on this channel. However, since channels are buffered, messages sent before the disconnect will still be properly received.

from https://doc.rust-lang.org/std/sync/mpsc/struct.Receiver.html#method.recv